examples/component/http-keyvalue-crud/main.go

```diff
@@ -0,0 +1,185 @@
+//go:generate go run go.bytecodealliance.org/cmd/wit-bindgen-go generate --world hello --out gen ./wit
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+
+	// A lightweight, high performance HTTP request router
+	"github.com/julienschmidt/httprouter"
+
+	// For the keyvalue capability, we're using bindings for the wasi:keyvalue/store interface.
+	store "github.com/wasmCloud/go/examples/component/http-keyvalue-crud/gen/wasi/keyvalue/store"
+
+	// The cm module provides types and functions for interacting with the WebAssembly Component Model.
+	"go.bytecodealliance.org/cm"
+
+	// The wasmCloud wasihttp module enables us to write more idiomatic Go when using wasi:http.
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+// Types for JSON validation.
+type CheckRequest struct {
+	Value string `json:"value"`
+}
+
+type CheckResponse struct {
+	Valid   bool   `json:"valid"`
+	Length  int    `json:"length,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+func init() {
+	// Establishes the routes and methods for our keyvalue operations.
+	router := httprouter.New()
+	router.HandlerFunc(http.MethodGet, "/", indexHandler)
+	router.HandlerFunc(http.MethodPost, "/post", postHandler)
+	router.HandlerFunc(http.MethodGet, "/get", getHandler)
+	router.HandlerFunc(http.MethodDelete, "/delete", deleteHandler)
+	wasihttp.Handle(router)
+}
+
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Provide a key as query string to /post, /get, or /delete with the corresponding method\n")
+}
+
+func postHandler(w http.ResponseWriter, r *http.Request) {
+	// Checks for a valid key as query string.
+	// For all routes and methods, the user will provide the key via a query string, for example:
+	// curl 'localhost:8000/get?key'
+	if len(r.URL.RawQuery) > 0 && regexp.MustCompile(`^[a-zA-Z0-9]*$`).MatchString(r.URL.RawQuery) {
```

Comment:
If you decide to use [named parameters](https://github.com/julienschmidt/httprouter?tab=readme-ov-file#named-parameters) from httprouter, then the need for this check will be completely obviated.